client/components/domains/use-your-domain-step/index.tsx

```diff
@@ -0,0 +1,316 @@
+									<Gridicon icon="checkmark" size={ 18 } />
+									{ phrase }
+								</div>
+							);
+						} ) }
+					</div>
+				</div>
+				<div className="use-your-domain-step__option-action">
+					<Button
+						className="use-your-domain-step__option-button"
+						primary={ isPrimary }
+						onClick={ onClick }
+					>
+						{ buttonText }
+					</Button>
+					<div className="use-your-domain-step__learn-more">{ learnMore }</div>
+				</div>
+			</div>
+		</Card>
+	);
+}
+
+type UseYourDomainStepProps = {
+	basePath: string;
+	goBack?: () => void;
+	initialQuery?: string;
+};
+
+function UseYourDomainStep( { basePath, goBack, initialQuery }: UseYourDomainStepProps ) {
+	const cartKey = useCartKey();
+	const { responseCart: cart } = useShoppingCart( cartKey );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const [ searchQuery ] = useState( initialQuery || '' );
+
+	const selectedSite = useSelector( getSelectedSite );
+	const productsList = useSelector( getProductsList );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const domainsWithPlansOnly = useSelector( ( state ) =>
+		currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+	);
+	const primaryWithPlansOnly = useSelector( ( state ) =>
+		currentUserHasFlag( state, NON_PRIMARY_DOMAINS_TO_FREE_USERS )
+	);
+
+	const domainsWithPlansOnlyButNoPlan =
+		domainsWithPlansOnly && selectedSite?.plan && ! isPlan( selectedSite.plan );
+	const productSlug = getDomainProductSlug( searchQuery );
+	const domainProductSalePrice = getDomainTransferSalePrice(
+		productSlug,
+		productsList,
+		currencyCode
+	);
+
+	let domainProductFreeText = null;
+	if ( isNextDomainFree( cart ) || isDomainBundledWithPlan( cart, searchQuery ) ) {
+		domainProductFreeText = translate( 'Free with your plan' );
+	} else if ( domainsWithPlansOnlyButNoPlan ) {
+		domainProductFreeText = translate( 'Included in paid plans' );
+	}
+
+	let transferSalePriceText = null;
+	if (
+		! isEmpty( domainProductSalePrice ) &&
+		! isNextDomainFree( cart ) &&
+		! isDomainBundledWithPlan( cart, searchQuery ) &&
+		domainsWithPlansOnlyButNoPlan
+	) {
+		transferSalePriceText = translate( 'Sale price is %(cost)s', {
+			args: { cost: domainProductSalePrice! },
+		} );
+	}
+
+	let transferPriceText = null;
+	const domainProductPrice = getDomainPrice( productSlug, productsList, currencyCode );
+	if ( domainProductPrice ) {
+		if (
+			isNextDomainFree( cart ) ||
+			isDomainBundledWithPlan( cart, searchQuery ) ||
+			domainsWithPlansOnlyButNoPlan ||
+			getDomainTransferSalePrice( productSlug, productsList, currencyCode )
+		) {
+			transferPriceText = translate( 'Renews at %(cost)s', { args: { cost: domainProductPrice } } );
+		} else {
+			transferPriceText = translate( '%(cost)s per year', { args: { cost: domainProductPrice } } );
+		}
+	}
+
+	let mappingPriceText;
+	const price = get( productsList, [ 'domain_map', 'cost' ], null );
```

Comment:
This is the only usage of `get` in this file and can be easily replaced with `?.`. Let's completely remove Lodash from here!

Reply:
Let's! https://github.com/Automattic/wp-calypso/pull/101854/commits/f52f737944224af3059dfe29f5ad3d8ff3c5dcdb